### PR TITLE
MX-374: Added GraphQL mutations and http clients for new Payments API.

### PR DESCRIPTION
**Pull request by maksym-krupenko**
# Jira Issue: [MX-374]

## What?
- New GraphQL mutations for new Payments API.

## Why?
- To decouple Payments API from signed state of user.

## Optional checklist
- [ ] Unit tests written
- [ ] Tested locally
- [ ] Codescouted

[MX-374]: https://hedvig.atlassian.net/browse/MX-374?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/api/upstreams/paymentService.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  fetchAvailablePaymentMethods(market: Market): Promise<AvailablePaymentMethodsResponse>
+}
+
+export const createPaymentServiceClient = (
+  client: HttpClient,
+): PaymentServiceClient => {
+  return {
+    connectPaymentDetails: async (onboardingId, request) => {
+      const suffix = onboardingId != null ? `?onboardingId=${onboardingId}` : ''
+      const response = await client.put(`${paymentDetailsUrlPrefix}${suffix}`, request)
```
Comment (cpiehl1) [inline src/api/upstreams/paymentService.ts:32]: This is a `post` now

File: src/resolvers/payment-connection/paymentConnection.graphqls [hunk 2]
```diff
@@ -0,0 +1,81 @@
+extend type Mutation {
+    paymentConnection_connectPayment(onboardingId: ID, input: ConnectPaymentInput!) : ConnectPaymentResult
```
Comment (cpiehl1) [inline src/resolvers/payment-connection/paymentConnection.graphqls:2]:
```suggestion
    paymentConnection_connectPayment(onboardingId: ID, input: ConnectPaymentInput!) : ConnectPaymentResult!
```

File: src/resolvers/payment-connection/paymentConnection.graphqls [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+union ConnectPaymentResult = PaymentResultCode | ActionRequired
+
+type PaymentResultCode {
+    resultCode: String
+}
+
+type ActionRequired {
+    action: CheckoutPaymentAction
```
Comment (cpiehl1) [inline src/resolvers/payment-connection/paymentConnection.graphqls:54]:
```suggestion
    action: CheckoutPaymentAction!
```

File: src/resolvers/payment-connection/paymentConnection.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      }
+      return parseResponse(connectPaymentResponse);
+    },
+    paymentConnection_submitAdditionalPaymentDetails: async (
+      _parent,
+      { id, input },
+      { upstream }
+    ): Promise<ConnectPaymentResult> => {
+      const submitAdditionalPayments = await upstream.paymentService.submitAdditionalPaymentDetails(id, input)
+      return parseResponse(submitAdditionalPayments)
```
Comment (cpiehl1) [inline src/resolvers/payment-connection/paymentConnection.ts:29]: We should probably update the onboarding session here if successful right?

File: src/resolvers/payment-connection/paymentConnection.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    },
+    paymentConnection_submitAdyenRedirection: async (
+      _parent,
+      { id, input },
+      { upstream }
+    ): Promise<PaymentResultCode> => {
+      const submitAdyenRedirection = await upstream.paymentService.submitAdyenRedirection(id, input)
+      return { ...submitAdyenRedirection }
```
Comment (cpiehl1) [inline src/resolvers/payment-connection/paymentConnection.ts:37]: Update onboarding here too

File: src/api/upstreams/paymentService.ts [hunk 6]
```diff
@@ -0,0 +1,174 @@
+import { HttpClient } from "../httpClient";
+import {
+  AdditionalPaymentDetailsInput,
+  ConnectPaymentInput, Market, Maybe, SubmitAdyenRedirectionInput
+} from "../../generated/graphql";
+
+export interface PaymentServiceClient {
+  connectPaymentDetails(
+    onboardingId: Maybe<string> | undefined,
```
Comment (cpiehl1) [inline src/api/upstreams/paymentService.ts:9]:
```suggestion
    onboardingId: Maybe<string>,
```
or

```suggestion
    onboardingId: string | null,
```

**Review by cpiehl1 — COMMENTED** [review]
Great stuff! Had some minor suggestions 😄

File: src/resolvers/payment-connection/paymentConnection.graphqls [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    pares: String!
+}
+
+extend type QuoteCart {
+    # new field added to attach payment data
+    paymentConnection: PaymentConnection
+}
+
+type PaymentConnection {
+    id: ID
```
Comment (cpiehl1) [inline src/resolvers/payment-connection/paymentConnection.graphqls:37]:
```suggestion
    id: ID!
```
right?

**Review by cpiehl1 — APPROVED** [review]
🚀

File: src/resolvers/payment-connection/paymentConnection.graphqls (hunk 7 above)
Comment (maksym-krupenko) [reply src/resolvers/payment-connection/paymentConnection.graphqls:37]: No, it's nullable for the case when payment connection is not connected, but possible payment methods are present

**Pull request merged**